node_shell.py

```diff
@@ -10,74 +10,241 @@
 More details in the README.md file.
 
 Author: Giovanni Pizzi, EPFL
+Contributor: Bonan Zhu
 
 Version: 0.1
 """
 import cmd2
+from cmd2 import ansi
 import functools
 from datetime import datetime
 import pytz
 import sys
+import re
+from traceback import print_exc
+from pprint import pformat
+from collections import namedtuple
+import contextlib
```

Comment:
Maybe this is a good point to add pre-commits. We can do it after this PR is merged, for simplicity (maybe you could open an issue on the repo, and I can then do it), in particular regarding sorting of imports.